### PR TITLE
Avoid losing options when removing the github.com permission

### LINKED ISSUE #2908: Extension options are reset when changing site access
When I change this to "When you click the extension", all my extension options get reset:
image

### PR DESCRIPTION
**Pull request by jack1142**
Closes #2908

### CODE REVIEW

File: source/options-storage.ts [hunk 1]
```diff
@@ -74,7 +74,8 @@ async function initializeAllOptions(): Promise<void> {
 	// Remove old domains
 	browser.permissions.onRemoved!.addListener(({origins}) => {
 		if (origins) {
-			const optionKeysToRemove = origins.map(origin => getStorageName(new URL(origin).host));
+			const optionKeysToRemove = origins.map(origin => getStorageName(new URL(origin).host))
+				.filter(key => key !== 'options');
```
Comment (fregante) [inline source/options-storage.ts]:
```suggestion
			const optionKeysToRemove = origins
				.filter(origin => origin !== 'https://github.com')
				.map(origin => getStorageName(new URL(origin).host));
```

Comment (jack1142) [reply source/options-storage.ts]: It doesn't - the actual origin would be `https://github.com/*` but still, it's not the only domain for which `getStorageName` can return `options` (`https://api.github.com/*`, `https://gist.github.com/*`)

Comment (fregante) [reply source/options-storage.ts]:
```suggestion
			const optionKeysToRemove = origins
				.map(origin => getStorageName(new URL(origin).host))
				.filter(key => key !== 'options');
```

**Comment by fregante** [comment]
Thank you! 🙌

**Pull request merged**